src/components/prescriptions/PrescriptionButtonComponent.vue

```diff
@@ -0,0 +1,284 @@
+                <span
+                    v-show="successFlag"
+                    aria-label="passwords match"
+                    class="dataSend"
+                    role="presentation"
+                    >Recepta została przesłana pomyślnie</span
+                >
+            </div>
+        </form>
+    </div>
+</template>
+
+<script setup>
+    import ButtonComponent from '../ButtonComponent.vue';
+    import BouncingBallsComponent from '../BouncingBallsComponent.vue';
+    import ErrorInfo from '../ErrorInfo.vue';
+    import getDoctorsArray from '../../composables/getDoctor.js';
+    import getMedicineArray from '../../composables/getMedicine.js';
+
+    import { computed, ref } from 'vue';
+    import ALLOWED_TYPES from '../../utils/allowedTypes.js';
+    import useStorage from '../../composables/getImage.js';
+    import useUserStore from '../../stores/userStore.js';
+    import { timestamp } from '../../firebase/config.js';
+    import useCollection from '../../composables/useCollections.js';
+    const { isLoading, addDoc, error } = useCollection('listOfRecipies');
+    const { url, uploadImage } = useStorage();
+
+    const addManually = ref(false);
+    const importFlag = ref(null);
+    const nameAndSurnameDoctor = ref(null);
+    const medicine = ref(null);
+    const frequency = ref(null);
+    const researchDate = ref(null);
+    const successFlag = ref(false);
+    const testFile = ref(null);
+    const testFileError = ref(null);
+
+    const userStore = useUserStore();
+
+    const disableAddTest = computed(
+        () => !(dateFormat.value || testFile.value)
+    );
+
+    const doctorNameAndSurname = computed(() => nameAndSurnameDoctor.value);
+
+    const medicineNames = computed(() => medicine.value);
+
+    const frequencyMedicines = computed(() => frequency.value);
+    const dateFormat = computed(
+        () =>
+            researchDate.value &&
+            researchDate.value.split('-').reverse().join('-')
+    );
+
+    const handleChange = (event) => {
+        const selectedFile = event.target.files[0];
+        if (selectedFile && ALLOWED_TYPES.includes(selectedFile.type)) {
+            testFile.value = selectedFile;
+            testFileError.value = null;
+        } else {
+            testFile.value = null;
+            testFileError.value =
+                'Please select an image file (jpeg, jpg, png, webp or pdf).';
+        }
+    };
+
+    const addTest = async () => {
+        try {
+            await uploadImage(testFile.value);
+            console.log(`image uploaded`, url.value);
+            await addDoc({
+                userId: userStore.getUserId,
+                date: dateFormat.value,
+                doctorData: doctorNameAndSurname.value,
+                testUrl: url.value,
+                medicineName: medicineNames.value,
+                frequencyMedicine: frequencyMedicines.value,
+                createdAt: timestamp(),
+            });
```

Comment:
there is problem with your handler cause if u want send either pdf / img nor object there is no any return statement or break and still if u send pdf your funtion also send object with nullish properties.